homeassistant/components/mqtt/config_flow.py

```diff
@@ -676,8 +615,8 @@ async def _async_validate_broker_settings(
         )
     ] = BOOLEAN_SELECTOR
     if (
-        current_client_certificate is None
-        and current_config.get(SET_CLIENT_CERT) is True
+        current_client_certificate is not None
+        or current_config.get(SET_CLIENT_CERT) is True
```

Comment:
We want to be able to upload client cert and key if the option `Set client cert` is set to `True` or if a client cert is already set up